### PR TITLE
meta: updated all markdown files metadata and updated files to be ready for translation

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

This PR normalises Markdown files and prepares certain contents for translation.

- All the blog pages were reformatted and had their Markdown frontmatter updated, removing obsolete frontmatter variables.
- Some pages, such as the Home Page, Docs Index had their content extracted directly to Markdown
- MDX Components got directly extracted to a separated `next.mdx.use.mjs` to define all shared MDX Components and HTML wirings
- Updated some translation keys and crowdin configuration
- Updated internal links using https://nodejs.org or /en/ to use simple `/` as we now prepend the current language.

## Validation

Pages should be behaving and rendering as expected. 

## Related Issues

Relates to https://github.com/nodejs/nodejs.org/issues/6036

### CODE REVIEW

File: pages/en/blog/release/v21.1.0.md [hunk 1]
```diff
@@ -1,9 +1,9 @@
 ---
-date: 2023-10-24T13:54:53.218Z
+date: '2023-10-24T13:54:53.218Z'
 category: release
 title: Node v21.1.0 (Current)
 layout: blog-post.hbs
-author: Michaël Zasso
+author: 'Michaël Zasso'
```
Comment (targos) [inline pages/en/blog/release/v21.1.0.md]: It's not clear if my name will be quoted in future release posts.

Comment (ovflowd) [reply pages/en/blog/release/v21.1.0.md]: It shouldn't; that's a good catch. I forgot to add special alphabetic characters to the regex. Ahem.

Comment (ovflowd) [reply pages/en/blog/release/v21.1.0.md]: Fixed.

**Review by AugustinMauroy — APPROVED** [review]

**Comment by ovflowd** [comment]
I'm fast-tracking this as it is mostly metadata update and fixing links. I've verified that links are working as much as I could... But @bmuenzenmeyer we might want to add a link checker workflow to run on `main` from time to time just to verify links are not broken.

**Pull request merged**